engine/src/silo/whitelist.rs

```diff
@@ -0,0 +1,156 @@
+use aurora_engine_sdk::io::{StorageIntermediate, IO};
+use aurora_engine_types::storage::{bytes_to_key, KeyPrefix};
+use aurora_engine_types::AsBytes;
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use crate::prelude::Vec;
+use crate::silo::parameters::{WhitelistKindArgs, WhitelistStatusArgs};
+
+const STATUS: &[u8] = b"LIST_STATUS";
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
+pub enum WhitelistKind {
+    Admin = 0x0,
+    EvmAdmin = 0x1,
+    Account = 0x2,
+    Address = 0x3,
+}
+
+impl From<WhitelistKind> for u8 {
+    fn from(list: WhitelistKind) -> Self {
+        match list {
+            WhitelistKind::Admin => 0x0,
+            WhitelistKind::EvmAdmin => 0x1,
+            WhitelistKind::Account => 0x2,
+            WhitelistKind::Address => 0x3,
+        }
+    }
+}
+
+/// `WhiteList` for checking access.
+/// * io - I/O trait handler
+pub struct WhiteList<I> {
```

Comment:
nit: whitelist is spelled as a single word so this should be `Whitelist` IMO.
```suggestion
pub struct Whitelist<I> {
```